lib/perl/Genome/Model/Build/ReferenceSequence/AnnotationIndex.pm

```diff
@@ -79,227 +41,33 @@ sub __display_name__ {
-    my $aligner_class = 'Genome::Model::Tools::'  . Genome::InstrumentData::AlignmentResult->_resolve_subclass_name_for_aligner_name($aligner_name);
-    return unless $aligner_class->can('supports_multiple_reference');
-    return $aligner_class->supports_multiple_reference($self->aligner_version);
-}
-
-sub get {
-    my $class = shift;
-
-    my @objects;
-    if (@_ % 2 == 0) {
-        my %p = @_;
-        unless ($p{test_name}) {
-            $p{test_name} = (Genome::Config::get('aligner_index_test_name') || undef);
-        }
-        if (exists $p{aligner_name} && $class->aligner_requires_param_masking($p{aligner_name})) {
-            $p{aligner_params} = undef;
-        }
-        @objects = $class->SUPER::get(%p);
-    } else {
-        @objects = $class->SUPER::get(@_);
-    }
-
-    return unless @objects;
-
-    for my $obj (@objects) {
-        next unless ref($obj); # sometimes UR gives us back the package name when deleting?
-
-        unless ($obj->check_dependencies()) {
-            $obj->error_message("Failed to get AnnotationIndex objects for dependencies of " . $obj->__display_name__);
-            return;
-        }
-    }
-
-    if (@objects > 1) {
-        return @objects if wantarray;
-        my @ids = map { $_->id } @objects;
-        die "Multiple matches for $class but get or create was called in scalar context! Found ids: @ids";
-    } else {
-        return $objects[0];
-    }
-}
-
-sub create {
-    my $class = shift;
-    my %p = @_;
-
-    unless ($p{test_name}) {
-        $p{test_name} = (Genome::Config::get('aligner_index_test_name') || undef);
-    }
-
-    my $aligner_class = 'Genome::InstrumentData::AlignmentResult::'  . Genome::InstrumentData::AlignmentResult->_resolve_subclass_name_for_aligner_name($p{aligner_name});
-    $class->debug_message("Aligner class name is $aligner_class");
-
-    $class->debug_message(sprintf("Resolved aligner class %s, making sure it's real and can be loaded.", $aligner_class));
-    unless ($aligner_class->class) {
-        $class->error_message(sprintf("Failed to load aligner class (%s).", $aligner_class));
-        return;
-    }
-
-    if ($class->aligner_requires_param_masking($p{aligner_name})) {
-        $p{aligner_params} = undef;
-    }
-
-    my $self = $class->SUPER::create(%p);
-    return unless $self;
-    $self->aligner_class_name($aligner_class);
-
-    $self->debug_message("Prepare staging directories...");
-    unless ($self->_prepare_staging_directory) {
-        $self->error_message("Failed to prepare working directory");
-        return;
-    }
-
-    unless ($self->_prepare_annotation_index) {
-        $self->error_message("Failed to prepare annotation index!");
-        return;
-    }
-
-    unless ($self->check_dependencies()) {
-        $self->error_message("Failed to create AnnotationIndex objects for dependencies");
-        return;
-    }
-
-    return $self;
-}
-
-sub check_dependencies {
+sub generate_dependencies_as_needed {
```

Comment:
The "dependency" thing was added for some particular aligner index, and none of our annotation indices work with it.  Until we have one that does, there's nothing to do...except maybe crash! :smile:   Perhaps this version of `generate_dependencies_as_needed` should go in the base class (in lieu of your `Carp::confess` suggestion)?